File: WebDriverTests_Java/WebDriverTests/src/test/java/core/systemControls/CheckBox.java
package core.systemControls;

import org.openqa.selenium.By;

/**
 * Created by Виктория on 03.12.2015.
 */
public class CheckBox  extends HtmlControl
{
    static
    {
    }


    public CheckBox(By locator)
    {
        this(locator, null);
    }

    public CheckBox(By locator, HtmlControl container)
    {
        super(locator, container);
    }
//
//    public final boolean getChecked()
//    {
//        return (boolean)element.getAttribute("checked");
//    }

//    public final void CheckAndWait()
//    {
//        if (!getChecked())
//        {
//            element.click();
//        }
//    }
//
//    public final void UncheckAndWait()
//    {
//        if (getChecked())
//        {
//            element.click();
//        }
//    }
//
//    public void WaitChecked()
//    {
//        String description = formatWithLocator(String.format("Ожидание Checked в элементе"));
//        Waiter.Wait(() -> getChecked() == true, description);
//    }
//
//    public void WaitUnchecked()
//    {
//        String description = formatWithLocator(String.format("Ожидание Checked в элементе"));
//        Waiter.Wait(() -> getChecked() == false, description);
//    }
}

File: WebDriverTests_Csharp/Autotest.Core/Autotests.Utilities/WebTestCore/SystemControls/CheckBox.cs
using System;
using Autotests.Utilities.WebTestCore.TestSystem;
using NUnit.Framework;
using OpenQA.Selenium;

namespace Autotests.Utilities.WebTestCore.SystemControls
{
    public class CheckBox : HtmlControl
    {
        static CheckBox()
        {
        }

        public CheckBox(By locator, HtmlControl container = null)
            : base(locator, container)
        {
        }

        public bool Checked
        {
            get { return Convert.ToBoolean(element.GetAttribute("checked")); }
        }

        public void CheckAndWait()
        {
            if (!Checked)
                element.Click();
        }

        public void UncheckAndWait()
        {
            if (Checked)
                element.Click();
        }

        public virtual void WaitChecked()
        {
            string description =
                FormatWithLocator(string.Format("Ожидание Checked в элементе"));
            Waiter.Wait(() => Checked == true, description);
        }

        public virtual void WaitUnchecked()
        {
            string description =
                FormatWithLocator(string.Format("Ожидание Checked в элементе"));
            Waiter.Wait(() => Checked == false, description);
        }
    }
}